pkg/agent/manager/pipe/pipe.go

```diff
@@ -0,0 +1,105 @@
+// TODO: ADD unit tests
+package pipe
```

Comment:
storepipe? storesvidpipe? do you have another ideas about names?

Reply:
What happens if a pipe is full?

I can't help but wonder if there are other ways to solve this problem - there's quite a bit of code present for managing these pipes

Reply:
thanks to your question, I found a bug I introduced when I was doing some tests... 
Right now pipe will lock (because chan) when limit reached, so no more svids will be pushed until we consume an element using `out`.

As you it is something that worries me... another thing we can do is put more logic on cache to push svids from there... Instead using buffered channels.
I avoid doing that because we'll adding a lot of logic there and I dont think we really want to make cache to use plugins in order to store SVIDs in some plattforms.